pkg/issuer/venafi/client/request.go

```diff
@@ -41,11 +42,44 @@ func (err ErrCustomFieldsType) Error() string {
 // This function sends a request to Venafi to for a signed certificate.
 // The CSR will be decoded to be validated against the zone configuration policy.
 // Upon the template being successfully defaulted and validated, the CSR will be sent, as is.
-func (v *Venafi) Sign(csrPEM []byte, duration time.Duration, customFields []internalvanafiapi.CustomField) (cert []byte, err error) {
+// It will return a pickup ID which can be used with RetreiveCertificate to get the certificate
+func (v *Venafi) RequestCertificate(csrPEM []byte, duration time.Duration, customFields []api.CustomField) (string, error) {
+	vreq, err := v.buildVReq(csrPEM, duration, customFields)
+	if err != nil {
+		return "", err
+	}
+	// Send the certificate signing request to Venafi
+	requestID, err := v.vcertClient.RequestCertificate(vreq)
+	return requestID, err
+}
+
+func (v *Venafi) RetreiveCertificate(pickupID string, csrPEM []byte, duration time.Duration, customFields []api.CustomField) ([]byte, error) {
+	vreq, err := v.buildVReq(csrPEM, duration, customFields)
+	if err != nil {
+		return nil, err
+	}
+
+	vreq.PickupID = pickupID
+	vreq.Timeout = time.Second * 10
```

Comment:
Before this was 5 minutes. I think 10 seconds makes a nice compromise for fast issuing certs to be available immediately, also this will block the controller only 10 seconds at a time. Also I cannot seem to find the official answer on this but it seems to also be the connection timeout